### DISCUSSION TITLE
特定のレイヤーでのみトラックボールを動作するようにすることはできますでしょうか？

### DISCUSSION THREAD

**Discussion by temosy** [Yowkees/keyball#138] [Q&A] [answered]
約一週間ほどKeyball44を使用しました。マウスを別途使用せずに仕事がおこなえることに感動し、手首の痛みが大幅に軽減されて大変満足しています。ありがとうございます！

自分自身がKeyball44に適応することで、以前使用していたキーボードの生産性を超えたいと思い、日々、楽しみながら試行錯誤しています。
一週間使用してみて、最もネックになっているポイントは現在どのレイヤーに自分が居るかを把握することだと感じました。(レイヤー間違いによる入力ミスが生産性を損ねている状況です)

＜避けたいこと＞
・頭の中でどのレイヤーに居るかを意識し続けること。(呼吸のように意識せず、これができれば良いのですが)
・OLEDの表示を見て確認すること。(タイピング時にキーボードに視線を向けるという不要なワンアクションは避けたいと考えております)

そこで、トラックボールが反応するかどうかで今X番のレイヤーに居るんだなと気づけるようにしたいのですが、そのようなことは可能でしょうか？

**Comment by Yowkees** [comment, marked as answer]
Keyball44のご愛用ありがとうございます！手首の痛みが軽減されたとのことで、お役に立てて大変光栄です。

さて、トラックボールの挙動によりレイヤーを確認するというアイデアについてです。
現在Keyballのレイヤーが切り替わったときのトラックボールの動作は、
レイヤー０→通常のトラックボール動作
レイヤー１→通常のトラックボール動作
レイヤー２→通常のトラックボール動作
レイヤー３→スクロール動作
となっております。
このうち、レイヤー１や２においてはトラックボールは使用できないようにするということでしょうか？
（理解が間違っていたらすみません。。）
そうなると左クリックを使用しながらカーソルを動かすことが難しくなる気がします。（レイヤー０に左クリックを配置できれば良いですが、キー数的に厳しいと思います。

もしそれでもトラックボール動作を止めたいということでしたら、ソフト的には可能ですのでご検討させていただきます。
ご確認よろしくお願いいたします。

**Reply by temosy** [reply]
ご返信ありがとうございます。まず、私がどのように使用しているかを説明するべきでした。

現状下記のような配置にしております。
[temosy-keymap_cheatsheet_keyball44.pdf](https://github.com/Yowkees/keyball/files/9690380/temosy-keymap_cheatsheet_keyball44.pdf)

そのうえで、0番と2番のレイヤーを行ったり来たりしています。それぞれのレイヤーの役割を下記のようにしています。
- 0番 文字入力
- 2番 数字、記号、マウス入力

この状況下で2番のレイヤー以外ではトラックボールを無効化したいです。
トラックボールを停止する方法があれば教えて頂けますと幸いです。

**Reply by temosy** [reply]
あれからさらに使い続け、RemapのHOLD/TAPの使い方を覚えました。
Shiftキーのように長押ししながら別のキーを押下し、別のレイヤーのキーとして動作させることができることに気づき、すべて自分の思い通りにすることに成功しました。

結論： RemapのHOLD/TAP で別レイヤーを割り当てれば、Shiftキーのように押しっぱなしにした時だけ別レイヤーにいることにできるから、どのレイヤーに自分が居るかわからなくなるレイヤー迷子現象を回避可能。🖲が反応するか否かで判断する必要すらない。

[temosy-keymap_cheatsheet_keyball44-fixed.pdf](https://github.com/Yowkees/keyball/files/10046700/temosy-keymap_cheatsheet_keyball44-fixed.pdf)

今まではレイヤーを活用することができていませんでしたが、Keyball44のおかげでレイヤーを意識するようになり、楽しみながらキー配置を考えることができました。もう完全に手放せない商売道具になりました。改めて素晴らしいキットをリリースしてくださりましてありがとうございます！
